internal/provider/resource_cloudflare_web3hostname.go

```diff
@@ -0,0 +1,92 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceCloudflareWeb3Hostname() *schema.Resource {
+	return &schema.Resource{
+		Schema:        resourceCloudflareWeb3HostnameSchema(),
+		CreateContext: resourceCloudflareWeb3HostnameCreate,
+		ReadContext:   resourceCloudflareWeb3HostnameRead,
+		UpdateContext: resourceCloudflareWeb3HostnameUpdate,
+		DeleteContext: resourceCloudflareWeb3HostnameDelete,
```

Comment:
this struct needs a `Description` method to update the overview section of the documentation. i.e. https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/internal/provider/resource_cloudflare_tunnel_route.go#L26-L30